hyper/http20/stream.py

```diff
@@ -94,8 +94,9 @@ def send_headers(self, end_stream=False):
         Sends the complete saved header block on the stream.
         """
         headers = self.get_headers()
-        self._conn.send_headers(self.stream_id, headers, end_stream)
-        self._send_cb(self._conn.data_to_send())
+        with self._conn as conn:
+            conn.send_headers(self.stream_id, headers, end_stream)
+        self._send_cb()
```

Comment:
Should we maybe rename the `_send_cb`? I'd like it to match the name as used in the connection object if possible.